include/eve/traits/overload/supports.hpp

```diff
@@ -0,0 +1,234 @@
+  };
+}
+
+namespace eve
+{
+  //====================================================================================================================
+  //! @addtogroup extensions
+  //! @{
+  //!   @struct supports
+  //!   @brief Helper class to aggregate options handlers
+  //!
+  //!   eve::callable can be decorated with options. eve::supports is used to list all subset of options a given EVE
+  //!   @callable is allowed to support and to handle the default values of such subsets.
+  //!
+  //!   @tparam Tag Type of current @callable being implemented
+  //!   @tparam Specs List of options specifications to support
+  //!
+  //!   @groupheader{Example}
+  //!
+  //!   @subgroupheader{Interaction with eve::conditional}
+  //!   @godbolt{doc/traits/callable_supports.cpp}
+  //!
+  //!   @subgroupheader{Interaction with custom specifications}
+  //!   @godbolt{doc/traits/callable_specs.cpp}
+  //! @}
+  //====================================================================================================================
+  template<typename Tag, typename... Specs>
+  struct supports
+  {
+    using is_decorated_callable = void;
+
+    // Aggregate all defaults from all Specs
+    EVE_FORCEINLINE static constexpr auto defaults() noexcept
+    {
+      return  kumi::fold_left ( [](auto const& acc, auto const& e) { return options{rbr::merge(e,acc)}; }
+                              , kumi::make_tuple(Specs::defaults()...)
+                              );
+    }
+
+    //==================================================================================================================
+    //! @brief Adds an option to current callable
+    //!
+    //! If the option `o` is supported by one of the specifications `Spec`, build a new @callable that will behave as
+    //! an instance of `Tag` except the option `o` is passed inside an eve::options as an additional parameter.
+    //!
+    //! Does not participate in overload resolution if `o` is not support by any option specifications.
+    //!
+    //! @param o Option to add to current's @callable setup
+    //! @return A eve::decorated_callable behaving as `Tag` but with the options `o` set.
+    //==================================================================================================================
+    template<detail::is_supported<Specs...> O>
+    EVE_FORCEINLINE auto operator[](O const& o) const
+    {
+      if constexpr(rbr::concepts::option<O>)
+      {
+        auto base = options{o};
+        auto opts = rbr::merge(base,defaults());
+        return detail::decorated_fn<Tag,decltype(options{opts}),Specs...>{opts};
+      }
+      else if constexpr(conditional_expr<O> || eve::logical_value<O>)
+      {
+        return (*this)[option::condition = o];
+      }
+      else if constexpr(std::same_as<O,bool>)
+      {
+        using type = std::conditional_t<std::same_as<bool,O>,std::uint8_t,O>;
+        return (*this)[option::condition = if_(logical<type>(o))];
+      }
```

Comment:
Same here - conditional should not be special

Reply:
they sort have to. Rabery can not jsut take a random value in [] if it is not associated to a keyword. The condition keyword is also required internally later to be able to say "I want to fetch the condition from the option bundle".

Now, and it is related to the other remark on [], I was rthinking of having the operator[] in condition to do this and generally have people write code to how to process values insiode a option bundle.

Reply:
Not really.  My problem in general with this design is that you taking away the logic from modifiers, give it back.

Modifiers should be the thing that determines what `operator[]` is provided.
Modifiers should be in charge of how to process them.

If I need to support a modifier of `int` - so be it.

Take options, return new options. See my `conditional_relative_in_out` example.